File: 0774-maximum-depth-of-n-ary-tree/0774-maximum-depth-of-n-ary-tree.java
/*
// Definition for a Node.
class Node {
    public int val;
    public List<Node> children;

    public Node() {}

    public Node(int _val) {
        val = _val;
    }

    public Node(int _val, List<Node> _children) {
        val = _val;
        children = _children;
    }
};
*/

class Solution {
    public int maxDepth(Node root) {
        if(root == null)return 0;
        return getMaxDepth(root);
    }
    int getMaxDepth(Node root){
        if(root == null)
            return 0;
        List<Node> children = root.children;
        int maxDepth = 0;
        for(int i=0; i<children.size(); i++)
            maxDepth = Math.max(maxDepth, getMaxDepth(children.get(i)));
        return 1+maxDepth;
    }
}

File: 0774-maximum-depth-of-n-ary-tree/0774-maximum-depth-of-n-ary-tree.py
"""
# Definition for a Node.
class Node:
    def __init__(self, val: Optional[int] = None, children: Optional[List['Node']] = None):
        self.val = val
        self.children = children
"""

class Solution:
    def maxDepth(self, root: 'Node') -> int:
        def getMaxDepth(node) -> int:
            if node is None:
                return 0
            maxi = 0
            children = node.children
            for child in children:
                maxi = max(maxi, getMaxDepth(child))
            return 1+maxi
        return getMaxDepth(root)
        